core/conn/odbc/src/odbc/nsksrvrcore/srvrothers.cpp

```diff
@@ -6533,6 +6533,8 @@ odbc_SQLSrvr_ExtractLob_sme_(
         if (retcode == SQL_ERROR)
         {
             ERROR_DESC_def *p_buffer = QryLobExtractSrvrStmt->sqlError.errorList._buffer;
+            char             errNumStr[128];
+            sprintf(errNumStr, "%d", p_buffer->sqlcode);
             strncpy(RequestError, p_buffer->errorText, sizeof(RequestError) - 1);
```

Comment:
Surround code issue.  RequestError may not be null terminated when the RequestError size is less than the length of the string in p_buffer->errorText. Also, this can cause core dump due to segment violation if length of errorText is less than the size of RequestBuffer.

Reply:
@selvaganesang  since RequestError is initialized at line 6489 and 6585, is there still a concern?
     char RequestError[200] = {0}; 

@xiaozhongwang if you want you could also take care of line 5254 ? 
Also do check regarding %d vs.  %ld for sqlcode.

Sorry for the delay in review. Thanks for the change.

Reply:
I don't look for this fix a long time. the branch have been deleted.
So I pull a new request for this, modified as your suggestion.